OpenHumans/open-humans
# rm legacy approach in public sharing model

## Description
This replaces a legacy approach to managing public data sharing, which has been using a "source label" approach. The model now refers directly to the project membership as a `OneToOneField`.

This is likely to change again if/when we work out how to use the new `DataTypes` in managing public data sharing.

This also makes some updates to `DataFile` and `ProjectDataFile`. (As part of this work I cleaned up some weirdnesses here as well. The presence of two models is a legacy behavior, as now every `DataFile` object should also have a `ProjectDataFile`.)

## Related Issue
This legacy approach in public data sharing status was encountered as I created a rewrite of the activity pages.

## Testing
  * passed automated testing locally (and updated existing testing to fix places it broke)
  * ran locally to test manually:
    * load activity pages
    * turn public data sharing off and on
    * ... and saw files disappear/appear in the API and profile page
    * quit/remove data from a project
    * list projects on "My data" page & am able to toggle there as well
